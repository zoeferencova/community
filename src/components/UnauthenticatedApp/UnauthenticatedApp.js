import React from "react";
import { Routes, Route } from "react-router-dom";
import { PropTypes } from 'prop-types';
import Nav from "../Nav/Nav";
import LandingPage from "../../routes/LandingPage/LandingPage";
import LoginPage from "../../routes/LoginPage/LoginPage";
import RegistrationPage from "../../routes/RegistrationPage/RegistrationPage";
import NotFoundPage from "../../routes/NotFoundPage/NotFoundPage";
import DeactivationSuccessPage from "../../routes/DeactivationSuccessPage/DeactivationSuccessPage";
import ErrorBoundary from "../ErrorBoundary/ErrorBoundary";

import styles from "./UnauthenticatedApp.module.css";

const UnauthenticatedApp = ({ isLoggedIn, setLoggedIn }) => (
  <main className={styles.main}>
    <Nav isLoggedIn={isLoggedIn} />
    <ErrorBoundary key={window.location.pathname}>
      <Routes>
        <Route exact path="/" element={<LandingPage setLoggedIn={setLoggedIn} isLoggedIn={isLoggedIn} />} />
        <Route path="/login" element={<LoginPage setLoggedIn={setLoggedIn} isLoggedIn={isLoggedIn} />} />
        <Route path="/register" element={<RegistrationPage />} />
        <Route path="/deactivated" element={<DeactivationSuccessPage />} />
        <Route element={<NotFoundPage isLoggedIn={isLoggedIn} />} />
      </Routes>
    </ErrorBoundary>
  </main>
)

export default UnauthenticatedApp;

UnauthenticatedApp.propTypes = {
  isLoggedIn: PropTypes.bool,
  setLoggedIn: PropTypes.func
}