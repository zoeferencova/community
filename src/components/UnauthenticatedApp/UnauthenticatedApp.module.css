.main {
    display: flex;
    flex-direction: column;
    height: 100%;
}

@media only screen and (min-width: 831px) {
    .authForm {
        background: #F7FAFA;
    }
}