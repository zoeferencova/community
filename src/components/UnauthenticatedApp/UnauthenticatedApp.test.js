import React from "react";
import { shallow } from "enzyme";
import UnauthenticatedApp from "./UnauthenticatedApp";


describe("UnauthenticatedApp", () => {
  it("renders UnauthenticatedApp component", () => {
    const wrapper = shallow(<UnauthenticatedApp />);

    expect(wrapper).toMatchSnapshot();
  });
});




