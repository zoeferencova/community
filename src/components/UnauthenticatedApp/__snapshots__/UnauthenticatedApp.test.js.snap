// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnauthenticatedApp renders UnauthenticatedApp component 1`] = `
<main
  className="main"
>
  <Nav />
  <withRouter(ErrorBoundary)
    key="/"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/login"
      />
      <Route
        component={[Function]}
        path="/register"
      />
      <Route
        component={[Function]}
        path="/deactivated"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </withRouter(ErrorBoundary)>
</main>
`;
